### PR TITLE
Add rethinkdb-python

### PR DESCRIPTION
**Pull request by jeongseok-meta**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by jeongseok-meta** [comment]
@conda-forge/help-python, ready for review!

File: recipes/rethinkdb-python/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  version: {{ version }}
+
+source:
+  url: https://pypi.org/packages/source/{{ name[0] }}/rethinkdb/rethinkdb-{{ version }}.tar.gz
+  sha256: 55ecf7447dc11fec0e2ce9aac4ba4c0b57fdc5c9c57d77ee673d59d245c74666
+
+build:
+  number: 0
+  script: {{ PYTHON }} -m pip install . -vv
+  skip: true  # [win]
```
Comment (xhochy) [inline recipes/rethinkdb-python/meta.yaml:15]: Why is this skipped? It looks like this could be a noarch package.

Comment (jeongseok-meta) [reply recipes/rethinkdb-python/meta.yaml:15]: The upstream package unfortunately does not support Windows (https://github.com/rethinkdb/rethinkdb-python/issues/217). I initially tried making it noarch, but this failed due to binary files and other issues. This PR doesn't include a detailed history of these failures, but I can reproduce them if needed.

Comment (jeongseok-meta) [reply recipes/rethinkdb-python/meta.yaml]: Sounds good to me. It's done in [77845af](https://github.com/conda-forge/staged-recipes/pull/28348/commits/77845af3812cc24175b0243e3b9199d3a745439c)!

Comment (jeongseok-meta) [reply recipes/rethinkdb-python/meta.yaml]: Hm, now win build fails with:

```
During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "D:\a\1\s\.ci_support\build_all.py", line 356, in <module>
    build_all(os.path.join(root_dir, "recipes"), args.arch)
  File "D:\a\1\s\.ci_support\build_all.py", line 185, in build_all
    build_folders(recipes_dir, folders, arch, channel_urls)
  File "D:\a\1\s\.ci_support\build_all.py", line 248, in build_folders
    conda_build.api.build([recipe], config=get_config(arch, channel_urls))
  File "D:\Miniforge\Lib\site-packages\conda_build\api.py", line 209, in build
    return build_tree(
           ^^^^^^^^^^^
  File "D:\Miniforge\Lib\site-packages\conda_build\build.py", line 3655, in build_tree
    packages_from_this = build(
                         ^^^^^^
  File "D:\Miniforge\Lib\site-packages\conda_build\build.py", line 2762, in build
    newly_built_packages = bundlers[pkg_type](
                           ^^^^^^^^^^^^^^^^^^^
  File "D:\Miniforge\Lib\site-packages\conda_build\build.py", line 1871, in bundle_conda
    files = post_process_files(metadata, initial_files)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Miniforge\Lib\site-packages\conda_build\build.py", line 1666, in post_process_files
    noarch_python.populate_files(
  File "D:\Miniforge\Lib\site-packages\conda_build\noarch_python.py", line 95, in populate_files
    handle_file(f, d, prefix)
  File "D:\Miniforge\Lib\site-packages\conda_build\noarch_python.py", line 76, in handle_file
    fn = rewrite_script(fn, prefix)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Miniforge\Lib\site-packages\conda_build\noarch_python.py", line 31, in rewrite_script
    raise CondaBuildUserError(f"Noarch package contains binary script: {fn}")
conda_build.exceptions.CondaBuildUserError: Noarch package contains binary script: rethinkdb-export.exe
```

Not sure if this means this package cannot be `noarch` or if there is a resolution that I am not aware of.

Comment (xhochy) [reply recipes/rethinkdb-python/meta.yaml]: You need to specify the Python entry_points as seen in https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#python-entry-points

File: recipes/rethinkdb-python/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.org/packages/source/{{ name[0] }}/rethinkdb/rethinkdb-{{ version }}.tar.gz
+  sha256: 55ecf7447dc11fec0e2ce9aac4ba4c0b57fdc5c9c57d77ee673d59d245c74666
+
+build:
+  number: 0
+  script: {{ PYTHON }} -m pip install . -vv
```
Comment (ocefpaf) [inline recipes/rethinkdb-python/meta.yaml:15]:
```suggestion
  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation 
```

File: recipes/rethinkdb-python/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,57 @@
+{% set name = "rethinkdb-python" %}
+{% set version = "2.4.10" %}
+{% set python_min = python_min|default("3.6") %}
```
Comment (xhochy) [inline recipes/rethinkdb-python/meta.yaml:3]: This should not be necessary with the most recent changes in staged-recipes.
```suggestion
```

**Pull request merged**